### PR TITLE
Miscellaneous notation fixes

### PR DESCRIPTION
**Pull request by dmitrio95**
Includes fixes for some common (mostly minor) musical notation issues.

### CODE REVIEW

File: libmscore/style.cpp [hunk 1]
```diff
@@ -1134,6 +1134,7 @@ static const StyleType styleTypes[] {
       { Sid::vibratoMinDistance,            "vibratoMinDistance",            Spatium(1.0)  },
       { Sid::voltaMinDistance,              "voltaMinDistance",              Spatium(1.0)  },
       { Sid::figuredBassMinDistance,        "figuredBassMinDistance",        Spatium(0.5)  },
+      { Sid::tupletMinDistance,             "tupletMinDistance",             Spatium(1.0)  },
```
Comment (MarcSabatella) [inline libmscore/style.cpp]: Isn't this a bit much?  It results in tuplets being further from notes than before.  I'd suggest 0.5 at most.

Comment (dmitrio95) [reply libmscore/style.cpp]: Yes, vtests actually show the same, I didn't notice this initially. It is much better visible when comparing 3.1 vs. 3.2 versions but comparing to the reference files also shows this difference, see http://vtest.musescore.org/e8a16cac/vtest.html#tuplets-1. So this should probably indeed be smaller. The only thing that may need checking is an appearance of tuplets above articulations, if it doesn't look good with the lower value of this setting then some additional spacing may be required for this case.

Comment (MarcSabatella) [reply libmscore/style.cpp]: It looks good in the cases I tried, so I have a commit reducing it in my PR #5189

Comment (dmitrio95) [reply libmscore/style.cpp]: Looking at vtests, the results of layout with this min. distance value still differ from those in 3.1 version. Reducing this value further reduces the differences but there is also another issue here that influences the result. The generic autoplacement algorithm uses the element's bounding box in order to find its proper placement related to a skyline. However this is not always correct for tuplets: tuplets with brackets have a more complex shape consisting of the bracket itself (which may have also some slope) and a number text. This makes the tuplet be sometimes unnecessarily pulled further from a staff because of the bracket edge that does not touch anything itself but makes a bounding box much higher. Calculating the distance using the element's shape instead of bbox in `autoplaceMeasureElement()` makes the differences from 3.1 version much smaller, and making this min. distance value equal to 0.3 or 0.4 makes only two vtest examples containing tuplets differ from their appearance in 3.1 version, and that difference is actually rather small (probably negligibly small). Tuplets above articulations still seem to look good with these min. distance values.

So, in conclusion, I believe the best option would be to use shapes for autoplacement for tuplets (that will work for all elements but, for performance reasons, it may be better to keep other elements using bboxes), possibly reducing the minimal autoplacement distance for tuplets to 0.4 or 0.3 of a spatium unit.

And a side question: do you think it would be useful (for anyone except me) to create something similar to [vtest.musescore.org](http://vtest.musescore.org/index.html) but with a possibility to show differences between different versions (and, possibly, vtest differences of certain pull request versus master/release versions)? I have modified vtest scripts to make it possible for me locally but this still doesn't make it easy to show the results to anyone else.

**Comment by MarcSabatella** [comment]
I agree about shapes rather than bounding boxes.  I noticed the same but just wanted to get a quick fix in, as I was under the impression there was still time to get this into 3.2.2  I think this needs to be an emergency fix one way or another.  Do you have some code using shapes ready to go?  I guess it could be as simple as special-casing tuplets in autoplaceMeasureElement, although maybe better is a new autoplaceShapeElement function that can be re-used for other element types.

**Comment by MarcSabatella** [comment]
As for vtests, we paid more attention to them in the run-up to 2.0 and in the first few updates after that.  So much changed with the "new layout" - that it became almost pointless to even look at results.  Now it's definitely worth taking some time to make this process easier.

**Comment by MarcSabatella** [comment]
Other than tuplets, autoplaceMeasureElement() is only used for jumps/markers.  I think we can afford to just use shapes everywhere here.

**Comment by MarcSabatella** [comment]
In my tests, when switching to shapes, a minDistance of 0.5sp is exactly the same as 0sp for all the vtests except for the second triplet in tuplets3, and that's because the right end of the bracket is very close to the flag on the last note.  I'm actually thinking 3.1 was placing this too close; using 0.5sp looks better to me.  And as soon as articulations enter the picture, 0.5sp is definitely better to my eyes than anything smaller.  So I still feel that's the best answer.

I'll update my PR #5189 as soon as I solve the problem of editing the bracket.  My initial approach of not doing the autoplace during the edit fails because, well, then there is no autoplace during the edit, so the bracket jumps back to its non-autoplaced position.  I see for spanners, we don't do a layout during editDarg(), just a triggerLayout().  I'm not seeing an obvious reason we can't do that here as well, still testing.
